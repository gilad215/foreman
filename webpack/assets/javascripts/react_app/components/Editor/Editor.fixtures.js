import { noop } from '../../common/helpers';

export const editor = {
  name: 'editor',
  diffViewType: 'split',
  value: 'value',
  hosts: { host1: 1, host2: 2 },
  renderedValue: '',
  mode: 'Ruby',
  theme: 'Monokai',
  keyBinding: 'Default',
  selectedView: 'input',
  editorName: 'editor',
  isMaximized: false,
  isMasked: false,
  isRendering: false,
  readOnly: false,
  showError: true,
  errorText: '',
};

export const dropdowns = {
  modes: ['Text', 'Json', 'Ruby', 'Html_ruby', 'Sh', 'Xml', 'Yaml'],
  keyBindings: ['Default', 'Emacs', 'Vim'],
  themes: ['Github', 'Monokai'],
};

export const showBooleans = {
  showPreview: true,
  showImport: true,
  showHide: true,
};

export const editorOptions = {
  ...editor,
  ...dropdowns,
  data: {
    ...showBooleans,
    name: 'editor',
    title: 'title',
    template: '<? />',
    options: dropdowns,
    hosts: [
      { host: { name: 'host1', id: 1 } },
      { host: { name: 'host2', id: 2 } },
    ],
  },
  initializeEditor: noop,
  importFile: noop,
  revertChanges: noop,
  previewTemplate: noop,
  toggleModal: noop,
  changeDiffViewType: noop,
  changeEditorValue: noop,
  dismissErrorToast: noop,
  changeTab: noop,
  toggleMaskValue: noop,
  changeSetting: noop,
  toggleRenderView: noop,
  template: '<? />',
  renderPath: '/render/path',
};

export const serverRenderResponse = {
  data: '< renderedData />',
};

export const hosts = [
  { host: { name: 'host1', id: 1 } },
  { host: { name: 'host2', id: 2 } },
];
