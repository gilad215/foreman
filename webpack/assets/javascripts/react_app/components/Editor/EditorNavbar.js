import React from 'react';
import PropTypes from 'prop-types';

import { ButtonGroup, Button, Dropdown, MenuItem, Icon } from 'patternfly-react';


const EditorNavbar = ({
  mode,
  theme,
  keyBinding,
  modes,
  themes,
  keyBindings,
  changeState,
  importFile,
  activeRadio,
  isDiff,
}) => (
  <div className="navbar navbar-form navbar-full-width navbar-editor">
    <ButtonGroup>
      <Button
        className={activeRadio === 'input' ? 'diff-button active' : 'diff-button'}
        id="input-btn"
        onClick={() => changeState('activeRadio', 'input')}
        bsStyle={activeRadio === 'input' ? 'primary' : 'default'}
      >
        Input
      </Button>
      <Button
        disabled={isDiff}
        className={activeRadio === 'diff' ? 'diff-button active' : 'diff-button'}
        id="diff-btn"
        onClick={() => changeState('activeRadio', 'diff')}
        bsStyle={activeRadio === 'diff' ? 'primary' : 'default'}
      >
        Diff
      </Button>
    </ButtonGroup>
    <div id="editor-dropdowns">
      <Dropdown id="mode-dropdown">
        <Dropdown.Toggle bsStyle="link">{mode}</Dropdown.Toggle>
        <Dropdown.Menu id="editor-dropdowns">
          <li className="dropdown-header">SYNTAX</li>
          <MenuItem divider />
          {modes.map((aceMode, i) => (
            <MenuItem key={i} onClick={() => changeState('mode', aceMode)}>
              {aceMode}
            </MenuItem>
          ))}
        </Dropdown.Menu>
      </Dropdown>
      <Dropdown id="keybindings-dropdown">
        <Dropdown.Toggle bsStyle="link">{keyBinding}</Dropdown.Toggle>
        <Dropdown.Menu id="editor-dropdowns">
          <li className="dropdown-header">KEYBIND</li>
          <MenuItem divider />
          {keyBindings.map((keyBind, i) => (
            <MenuItem key={i} onClick={() => changeState('keyBinding', keyBind)}>
              {keyBind}
            </MenuItem>
          ))}
        </Dropdown.Menu>
      </Dropdown>
      <Dropdown id="themes-dropdown">
        <Dropdown.Toggle bsStyle="link">{theme}</Dropdown.Toggle>
        <Dropdown.Menu id="editor-dropdowns">
          <li className="dropdown-header">THEME</li>
          <MenuItem divider />
          {themes.map((themeKey, i) => (
            <MenuItem key={i} onClick={() => changeState('theme', themeKey)}>
              {themeKey}
            </MenuItem>
          ))}
        </Dropdown.Menu>
      </Dropdown>
      <Button
        className='diff-button'
        id="import-btn"
        // onClick={() => changeState('showModal', true)}
        bsStyle='link'
      >
      <Icon size="lg" type="fa" name ="download" />
      </Button>
      <Button
        className='diff-button'
        id="fullscreen-btn"
        onClick={() => changeState('showModal', true)}
        bsStyle='link'
      >
      <Icon size="lg" type="fa" name ="arrows-alt" />
      </Button>
    </div>
  </div>
);

EditorNavbar.propTypes = {
  mode: PropTypes.string.isRequired,
  theme: PropTypes.string.isRequired,
  keyBinding: PropTypes.string.isRequired,
  modes: PropTypes.array.isRequired,
  themes: PropTypes.array.isRequired,
  keyBindings: PropTypes.array.isRequired,
  changeState: PropTypes.func.isRequired,
  importFile: PropTypes.func.isRequired,
  activeRadio: PropTypes.string.isRequired,
};

export default EditorNavbar;
