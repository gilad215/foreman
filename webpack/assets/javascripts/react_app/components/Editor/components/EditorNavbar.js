import React from 'react';
import PropTypes from 'prop-types';
import { Nav } from 'patternfly-react';
import { translate as __ } from '../../../common/I18n';

import EditorRadioButton from './EditorRadioButton';
import EditorOptions from './EditorOptions';

const EditorNavbar = ({
  value,
  mode,
  hosts,
  renderPath,
  template,
  theme,
  keyBinding,
  modes,
  themes,
  diffViewType,
  keyBindings,
  changeState,
  toggleModal,
  revertChanges,
  showPreview,
  previewTemplate,
  showHide,
  showImport,
  importFile,
  selectedView,
  isDiff,
  isMasked,
  isRendering,
}) => (
  <div className="navbar navbar-form navbar-full-width navbar-editor">
    <Nav className="nav nav-tabs nav-tabs-pf nav-tabs-pf-secondary">
      <EditorRadioButton
        stateView={selectedView}
        btnView="input"
        title={__('Editor')}
        onClick={() => {
          if (selectedView !== 'input') changeState({ selectedView: 'input' });
          if (isRendering) changeState({ isRendering: false });
        }}
      />
      {showPreview && (
        <EditorRadioButton
          stateView={selectedView}
          btnView="preview"
          title={__('Preview')}
          onClick={() => {
            if (selectedView !== 'preview')
              changeState({ selectedView: 'preview' });
            if (!isRendering) changeState({ isRendering: true });
          }}
        />
      )}
      <EditorRadioButton
        stateView={selectedView}
        disabled={!isDiff}
        btnView="diff"
        title={__('Changes')}
        onClick={() => {
          if (selectedView !== 'diff') changeState({ selectedView: 'diff' });
          if (isRendering) changeState({ isRendering: false });
        }}
      />
    </Nav>
    <EditorOptions
      hosts={hosts}
      value={value}
      renderPath={renderPath}
      previewTemplate={previewTemplate}
      showImport={showImport}
      showHide={showHide}
      showPreview={showPreview}
      isDiff={isDiff}
      diffViewType={diffViewType}
      isMasked={isMasked}
      isRendering={isRendering}
      importFile={importFile}
      template={template}
      revertChanges={revertChanges}
      changeState={changeState}
      toggleModal={toggleModal}
      selectedView={selectedView}
      mode={mode}
      modes={modes}
      keyBinding={keyBinding}
      keyBindings={keyBindings}
      theme={theme}
      themes={themes}
    />
  </div>
);

EditorNavbar.propTypes = {
  value: PropTypes.string.isRequired,
  mode: PropTypes.string.isRequired,
  theme: PropTypes.string.isRequired,
  template: PropTypes.string,
  keyBinding: PropTypes.string.isRequired,
  modes: PropTypes.array.isRequired,
  themes: PropTypes.array.isRequired,
  keyBindings: PropTypes.array.isRequired,
  changeState: PropTypes.func.isRequired,
  toggleModal: PropTypes.func.isRequired,
  renderPath: PropTypes.string,
  revertChanges: PropTypes.func.isRequired,
  importFile: PropTypes.func.isRequired,
  previewTemplate: PropTypes.func.isRequired,
  hosts: PropTypes.object,
  selectedView: PropTypes.string.isRequired,
  showPreview: PropTypes.bool.isRequired,
  showImport: PropTypes.bool.isRequired,
  showHide: PropTypes.bool,
  isMasked: PropTypes.bool.isRequired,
  isDiff: PropTypes.bool.isRequired,
  isRendering: PropTypes.bool.isRequired,
  diffViewType: PropTypes.string.isRequired,
};

EditorNavbar.defaultProps = {
  showHide: false,
  template: '',
  hosts: {},
  renderPath: '',
};

export default EditorNavbar;
