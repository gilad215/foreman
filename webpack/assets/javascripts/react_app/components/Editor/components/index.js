import { Loadable } from '../../../utils';

export default Loadable({
  loader: () => import('./Editor'),
});
