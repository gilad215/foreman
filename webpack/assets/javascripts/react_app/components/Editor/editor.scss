#editor-dropdowns {
  float: right;

  #mode-dropdown, #keybindings-dropdown, #themes-dropdown {
    min-width: 85px;
    margin-right: 5px;
  }

  #editor-dropdowns {
    min-width: 85px;
    text-align: center;
  }
}

.diff-button {
  width: 60px;
  margin-right: 2px;
}

#fullscreen-btn, #import-btn {
  width: 40px !important;
}

.ace_editor_form {
  position: relative !important;
  height: 360px !important;
  width: 100% !important;
}

.ace_editor_modal {
  width: 100% !important;
  height: -webkit-fill-available !important;
}

.editor-modal {
  .modal-content { 
    height: -webkit-fill-available;
  }
  .modal-dialog {
    padding: 15px;
    width: auto !important;

    .input.modal-body {
      background-color: #363636;
    }

    .modal-body {
      padding: 0;

      .ace_editor {
        border: none;
      }

      #diff-container {
        height: 100%;

        #diff-table {
          max-height: none;
          border: none;
        }
      }
    }

    .input.modal-header.monokai {
      background-color: #363636;

      .pficon-close {
        color: #fff;
      }
    }
  }
}