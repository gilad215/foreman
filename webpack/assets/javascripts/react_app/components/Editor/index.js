import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import * as actions from './EditorActions';
import reducer from './EditorReducer';

import Editor from './components';

// map state to props
const mapStateToProps = ({ editor }) => ({
  value: editor.value,
  hosts: editor.hosts,
  renderedValue: editor.renderedValue,
  errorText: editor.errorText,
  mode: editor.mode,
  theme: editor.theme,
  keyBinding: editor.keyBinding,
  selectedView: editor.selectedView,
  editorName: editor.editorName,
  diffViewType: editor.diffViewType,
  isMaximized: editor.isMaximized,
  isMasked: editor.isMasked,
  isRendering: editor.isRendering,
  readOnly: editor.readOnly,
  showError: editor.showError,
});

// map action dispatchers to props
const mapDispatchToProps = dispatch => bindActionCreators(actions, dispatch);

// export reducers
export const reducers = { editor: reducer };

// export connected component
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Editor);
