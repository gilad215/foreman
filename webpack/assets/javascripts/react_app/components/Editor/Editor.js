import React from 'react';
import PropTypes from 'prop-types';
import { ToastNotification } from 'patternfly-react';

import { noop } from '../../common/helpers';
import DiffView from '../DiffView/DiffView';
import EditorView from './components/EditorView';
import EditorNavbar from './components/EditorNavbar';
import EditorModal from './components/EditorModal';
import {
  EDITOR_THEMES,
  EDITOR_KEYBINDINGS,
  EDITOR_MODES,
} from './EditorConstants';
import './editor.scss';

class Editor extends React.Component {
  componentDidMount() {
    const {
      data: { hosts, locked, template, type },
      initializeEditor,
      isMasked,
      isRendering,
      readOnly,
      renderedValue,
      selectedView,
      showError,
    } = this.props;

    const initializeData = {
      hosts,
      isMasked,
      isRendering,
      locked,
      readOnly,
      renderedValue,
      selectedView,
      showError,
      template,
      type,
    };
    initializeEditor(initializeData);
  }

  render() {
    const {
      data: {
        name,
        renderPath,
        showHide,
        showImport,
        showPreview,
        template,
        title,
      },
      changeDiffViewType,
      changeEditorValue,
      changeSetting,
      changeTab,
      diffViewType,
      dismissErrorToast,
      editorName,
      errorText,
      hosts,
      importFile,
      isMasked,
      isMaximized,
      isRendering,
      keyBinding,
      mode,
      previewTemplate,
      readOnly,
      renderedValue,
      revertChanges,
      selectedView,
      showError,
      theme,
      toggleMaskValue,
      toggleModal,
      toggleRenderView,
      value,
    } = this.props;

    return (
      <div id="editor-container">
        <EditorNavbar
          changeDiffViewType={changeDiffViewType}
          changeTab={changeTab}
          changeSetting={changeSetting}
          modes={EDITOR_MODES}
          themes={EDITOR_THEMES}
          keyBindings={EDITOR_KEYBINDINGS}
          mode={isRendering ? 'Text' : mode}
          theme={theme}
          keyBinding={keyBinding}
          value={value}
          diffViewType={diffViewType}
          template={template}
          selectedView={selectedView}
          isDiff={template ? value !== template : false}
          isMasked={isMasked}
          isRendering={isRendering}
          importFile={importFile}
          showImport={showImport}
          showPreview={showPreview}
          showHide={showHide}
          revertChanges={revertChanges}
          previewTemplate={previewTemplate}
          hosts={hosts}
          renderPath={renderPath}
          toggleMaskValue={toggleMaskValue}
          toggleRenderView={toggleRenderView}
          toggleModal={toggleModal}
        />
        <ToastNotification
          id="preview_error_toast"
          type="error"
          className={showError ? '' : 'hidden'}
          onDismiss={() => dismissErrorToast()}
        >
          {errorText}
        </ToastNotification>
        <EditorView
          key="editorView"
          value={isRendering ? renderedValue : value}
          name={editorName}
          mode={isRendering ? 'Text' : mode}
          theme={theme}
          keyBinding={keyBinding}
          onChange={isRendering ? noop : changeEditorValue}
          readOnly={readOnly || isRendering}
          className={selectedView !== 'diff' ? 'ace_editor_form' : 'hidden'}
          isMasked={isMasked}
        />
        <div
          id="diff-table"
          className={selectedView === 'diff' ? '' : 'hidden'}
        >
          <DiffView
            oldText={template || ''}
            newText={value}
            viewType={diffViewType}
          />
        </div>
        <EditorModal
          key="editorModal"
          changeEditorValue={changeEditorValue}
          changeDiffViewType={changeDiffViewType}
          name={editorName}
          title={title}
          toggleModal={toggleModal}
          diffViewType={diffViewType}
          mode={isRendering ? 'Text' : mode}
          theme={theme}
          keyBinding={keyBinding}
          readOnly={readOnly}
          isMaximized={isMaximized}
          template={template || ''}
          editorValue={isRendering ? renderedValue : value}
          selectedView={selectedView}
          isMasked={isMasked}
        />
        <textarea className="hidden" name={name} value={value} readOnly />
      </div>
    );
  }
}

Editor.propTypes = {
  data: PropTypes.shape({
    showHide: PropTypes.bool,
    showImport: PropTypes.bool,
    showPreview: PropTypes.bool,
    template: PropTypes.string,
    name: PropTypes.string,
    title: PropTypes.string,
    renderPath: PropTypes.string,
  }).isRequired,
  changeDiffViewType: PropTypes.func.isRequired,
  changeEditorValue: PropTypes.func.isRequired,
  changeSetting: PropTypes.func.isRequired,
  changeTab: PropTypes.func.isRequired,
  diffViewType: PropTypes.string.isRequired,
  dismissErrorToast: PropTypes.func.isRequired,
  editorName: PropTypes.string.isRequired,
  errorText: PropTypes.string.isRequired,
  hosts: PropTypes.object.isRequired,
  importFile: PropTypes.func.isRequired,
  initializeEditor: PropTypes.func.isRequired,
  isMasked: PropTypes.bool.isRequired,
  isMaximized: PropTypes.bool.isRequired,
  isRendering: PropTypes.bool.isRequired,
  keyBinding: PropTypes.string.isRequired,
  mode: PropTypes.string.isRequired,
  previewTemplate: PropTypes.func.isRequired,
  readOnly: PropTypes.bool.isRequired,
  renderedValue: PropTypes.string.isRequired,
  revertChanges: PropTypes.func.isRequired,
  selectedView: PropTypes.string.isRequired,
  showError: PropTypes.bool.isRequired,
  theme: PropTypes.string.isRequired,
  toggleMaskValue: PropTypes.func.isRequired,
  toggleModal: PropTypes.func.isRequired,
  toggleRenderView: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
};

export default Editor;
