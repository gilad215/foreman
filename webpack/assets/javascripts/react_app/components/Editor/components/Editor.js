import React from 'react';
import PropTypes from 'prop-types';
import { ToastNotification } from 'patternfly-react';

import { noop } from '../../../common/helpers';
import DiffView from '../../DiffView/DiffView';
import EditorView from './EditorView';
import EditorNavbar from './EditorNavbar';
import EditorModal from './EditorModal';
import '../editor.scss';

class Editor extends React.Component {
  componentDidMount() {
    const { initializeEditor } = this.props;
    initializeEditor(this.props);
  }

  render() {
    const {
      data: {
        showHide,
        showImport,
        showPreview,
        template,
        name,
        renderPath,
        options,
        title,
      },
      value,
      hosts,
      mode,
      theme,
      keyBinding,
      selectedView,
      renderedValue,
      editorName,
      diffViewType,
      changeState,
      importFile,
      revertChanges,
      previewTemplate,
      toggleModal,
      readOnly,
      showError,
      errorText,
      isMasked,
      isMaximized,
      isRendering,
    } = this.props;

    return (
      <div id="editor-container">
        <EditorNavbar
          value={value}
          diffViewType={diffViewType}
          template={template}
          mode={isRendering ? 'Text' : mode}
          theme={theme}
          keyBinding={keyBinding}
          selectedView={selectedView}
          modes={options.modes}
          themes={options.themes}
          keyBindings={options.keyBindings}
          changeState={changeState}
          toggleModal={toggleModal}
          showImport={showImport}
          showPreview={showPreview}
          showHide={showHide}
          importFile={importFile}
          revertChanges={revertChanges}
          previewTemplate={previewTemplate}
          hosts={hosts}
          renderPath={renderPath}
          isDiff={template ? value !== template : false}
          isMasked={isMasked}
          isRendering={isRendering}
        />
        <ToastNotification
          id="preview_error_toast"
          type="error"
          className={showError ? '' : 'hidden'}
          onDismiss={() => changeState({ showError: false })}
        >
          {errorText}
        </ToastNotification>
        <EditorView
          key="editorView"
          value={isRendering ? renderedValue : value}
          name={editorName}
          mode={isRendering ? 'Text' : mode}
          theme={theme}
          keyBinding={keyBinding}
          onChange={isRendering ? noop : changeState}
          readOnly={readOnly || isRendering}
          className={selectedView !== 'diff' ? 'ace_editor_form' : 'hidden'}
          isMasked={isMasked}
        />
        <div
          id="diff-table"
          className={selectedView === 'diff' ? '' : 'hidden'}
        >
          <DiffView
            oldText={template || ''}
            newText={value}
            viewType={diffViewType}
          />
        </div>
        <EditorModal
          key="editorModal"
          changeState={changeState}
          name={editorName}
          title={title}
          toggleModal={toggleModal}
          diffViewType={diffViewType}
          mode={isRendering ? 'Text' : mode}
          theme={theme}
          keyBinding={keyBinding}
          readOnly={readOnly}
          isMaximized={isMaximized}
          template={template || ''}
          editorValue={isRendering ? renderedValue : value}
          selectedView={selectedView}
          isMasked={isMasked}
        />
        <textarea className="hidden" name={name} value={value} readOnly />
      </div>
    );
  }
}

Editor.propTypes = {
  data: PropTypes.shape({
    showHide: PropTypes.bool,
    showImport: PropTypes.bool,
    showPreview: PropTypes.bool,
    template: PropTypes.string,
    name: PropTypes.string,
    title: PropTypes.string,
    renderPath: PropTypes.string,
    options: PropTypes.shape({
      modes: PropTypes.arrayOf(PropTypes.string),
      keyBindings: PropTypes.arrayOf(PropTypes.string),
      themes: PropTypes.arrayOf(PropTypes.string),
    }).isRequired,
  }).isRequired,
  changeState: PropTypes.func.isRequired,
  importFile: PropTypes.func.isRequired,
  revertChanges: PropTypes.func.isRequired,
  previewTemplate: PropTypes.func.isRequired,
  initializeEditor: PropTypes.func.isRequired,
  toggleModal: PropTypes.func.isRequired,
  renderedValue: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  hosts: PropTypes.object.isRequired,
  editorName: PropTypes.string.isRequired,
  diffViewType: PropTypes.string.isRequired,
  mode: PropTypes.string.isRequired,
  theme: PropTypes.string.isRequired,
  keyBinding: PropTypes.string.isRequired,
  selectedView: PropTypes.string.isRequired,
  readOnly: PropTypes.bool.isRequired,
  showError: PropTypes.bool.isRequired,
  errorText: PropTypes.string.isRequired,
  isMasked: PropTypes.bool.isRequired,
  isMaximized: PropTypes.bool.isRequired,
  isRendering: PropTypes.bool.isRequired,
};

export default Editor;
