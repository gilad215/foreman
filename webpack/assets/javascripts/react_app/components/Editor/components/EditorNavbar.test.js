import React from 'react';
import { mount } from 'enzyme';
import { testComponentSnapshotsWithFixtures } from '../../../common/testHelpers';

import EditorNavbar from './EditorNavbar';
import { editorOptions, showBooleans } from '../Editor.fixtures';

const props = {
  ...editorOptions,
  ...showBooleans,
  isDiff: true,
};

const fixtures = {
  'renders EditorNavbar': props,
};

describe('EditorNavbar', () => {
  describe('rendring', () =>
    testComponentSnapshotsWithFixtures(EditorNavbar, fixtures));

  describe('simulate onClick', () => {
    const changeState = jest.fn();

    const wrapper = mount(
      <EditorNavbar
        {...props}
        changeState={changeState}
        isDiff
        isRendering
        selectedView="preview"
      />
    );
    wrapper
      .find('#input-navitem')
      .at(1)
      .simulate('click');
    wrapper
      .find('#diff-navitem')
      .at(1)
      .simulate('click');

    wrapper.setProps({ ...props, isRendering: false, selectedView: 'input' });
    wrapper.update();
    wrapper
      .find('#preview-navitem')
      .at(1)
      .simulate('click');
    expect(changeState).toHaveBeenCalledTimes(4);
  });
});
