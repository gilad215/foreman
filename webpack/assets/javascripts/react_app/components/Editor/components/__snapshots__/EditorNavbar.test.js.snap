// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorNavbar rendring renders EditorNavbar 1`] = `
<div
  className="navbar navbar-form navbar-full-width navbar-editor"
>
  <Nav
    bsClass="nav"
    className="nav nav-tabs nav-tabs-pf nav-tabs-pf-secondary"
    justified={false}
    pullLeft={false}
    pullRight={false}
    stacked={false}
  >
    <EditorRadioButton
      btnView="input"
      disabled={false}
      icon={null}
      onClick={[Function]}
      stateView="input"
      title="Editor"
    />
    <EditorRadioButton
      btnView="preview"
      disabled={false}
      icon={null}
      onClick={[Function]}
      stateView="input"
      title="Preview"
    />
    <EditorRadioButton
      btnView="diff"
      disabled={false}
      icon={null}
      onClick={[Function]}
      stateView="input"
      title="Changes"
    />
  </Nav>
  <EditorOptions
    changeState={[Function]}
    diffViewType="split"
    hosts={
      Object {
        "host1": 1,
        "host2": 2,
      }
    }
    importFile={[Function]}
    isDiff={true}
    isMasked={false}
    isRendering={false}
    keyBinding="Default"
    keyBindings={
      Array [
        "Default",
        "Emacs",
        "Vim",
      ]
    }
    mode="Ruby"
    modes={
      Array [
        "Text",
        "Json",
        "Ruby",
        "Html_ruby",
        "Sh",
        "Xml",
        "Yaml",
      ]
    }
    previewTemplate={[Function]}
    renderPath="/render/path"
    revertChanges={[Function]}
    selectedView="input"
    showHide={true}
    showImport={true}
    showPreview={true}
    template="<? />"
    theme="Monokai"
    themes={
      Array [
        "Github",
        "Monokai",
      ]
    }
    toggleModal={[Function]}
    value="value"
  />
</div>
`;
